rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -288,222 +258,146 @@ public String defineNodeSource(final String sessionId, final String nodeSourceNa
-     * java.util.Set)
-     */
-    public String lockNodes(final String sessionId, Set<String> urls) throws RestServerException, ServiceException {
-        return executeFunction(new BiFunction<RestClient, Set<String>, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient, Set<String> strings) {
-                return restClient.lockNodes(sessionId, strings);
-            }
-        }, urls, "lock");
+    @Override
+    public String getNodeSourceConfiguration(String sessionId, String nodeSourceName)
+            throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.getNodeSourceConfiguration(sessionId,
+                                                                                                       nodeSourceName));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#unlockNodes(java.lang.String,
-     * java.util.Set)
-     */
-    public String unlockNodes(final String sessionId, Set<String> urls) throws RestServerException, ServiceException {
-        return executeFunction(new BiFunction<RestClient, Set<String>, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient, Set<String> strings) {
-                return restClient.unlockNodes(sessionId, strings);
-            }
-        }, urls, "unlock");
+    @Override
+    public String lockNodes(String sessionId, Set<String> urls) throws RestServerException, ServiceException {
+        return executeFunction((restClient, strings) -> restClient.lockNodes(sessionId, strings), urls, "lock");
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#removeNode(java.lang.String,
-     * java.lang.String)
-     */
-    public String removeNode(final String sessionId, final String url, final boolean force)
-            throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.removeNode(sessionId, url, force);
-            }
-        });
+    @Override
+    public String unlockNodes(String sessionId, Set<String> urls) throws RestServerException, ServiceException {
+        return executeFunction((restClient, strings) -> restClient.unlockNodes(sessionId, strings), urls, "unlock");
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * org.ow2.proactive_grid_cloud_portal.rm.client.RMService#removeNodesource(java.lang.String,
-     * java.lang.String)
-     */
-    public String removeNodesource(final String sessionId, final String name, final boolean preempt)
+    @Override
+    public String removeNode(String sessionId, String url, boolean force) throws RestServerException, ServiceException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)